extensions/reactive-pg-client/deployment/src/main/java/io/quarkus/reactive/pg/client/deployment/ReactivePgClientProcessor.java

```diff
@@ -121,6 +129,23 @@ void addHealthCheck(
                         dataSourcesBuildTimeConfig.healthEnabled));
     }
 
+    @BuildStep
+    void unremoveableBeans(BuildProducer<UnremovableBeanBuildItem> producer) {
+        producer.produce(UnremovableBeanBuildItem.beanTypes(PgPoolCreator.class));
+    }
+
+    @BuildStep
+    void validateBeans(ValidationPhaseBuildItem validationPhase,
+            BuildProducer<ValidationPhaseBuildItem.ValidationErrorBuildItem> errors) {
+        long pgPoolCreatorCount = validationPhase.getContext()
+                .beans().matchBeanTypes(new PgPoolCreatorBeanClassPredicate()).stream().count();
+        if (pgPoolCreatorCount > 1) {
```

Comment:
Will this hold true if we have multiple creator instances for different, named datasources (with qualifiers?)

Reply:
With the current approach, it does. If we switch to qualifiers, we will need a smarter check